src/config/containers-and-vms/chroot.md

```diff
@@ -0,0 +1,90 @@
+# Creating and using chroots and containers
+
+chroots and containers can be set up and used for many purposes, including:
+
+- running glibc software on musl (and vice versa)
+- running software in a more controlled or sandboxed environment
+- creating a rootfs for bootstrapping a system
+
+## Chroot Creation
+
+Create a directory that will contain the chroot, then install a base system in
+it via the `base-voidstrap` package:
+
+```
+$ XBPS_TARGET_ARCH=<chroot_arch> xbps-install -M -r <chroot_dir> -R <repository> base-voidstrap
+```
+
+The `<repository>` varies depending on architecture:
+
+- `aarch64`, `aarch64-musl`:
+   `https://repo-default.voidlinux.org/current/aarch64`
+- any other `musl` architecture:
+   `https://repo-default.voidlinux.org/current/musl`
+- any other architecture: `https://repo-default.voidlinux.org/current`
+
+Alternatively, this can be done automatically with
+[`xvoidstrap(1)`](https://man.voidlinux.org/xvoidstrap.1) (from `xtools`):
+
+```
+# XBPS_ARCH=<chroot_arch> xvoidstrap <chroot_dir> base-voidstrap
+```
+
+## Chroot Usage
+
+### xchroot
+
+[`xchroot(1)`](https://man.voidlinux.org/xchroot.1) (from `xtools`) can be used
+to automatically set up and enter the chroot.
+
+### Manual Method
+
+Alternatively, this process can be done manually.
+
+If network access is required, copy `/etc/resolv.conf` into the chroot;
+`/etc/hosts` may need to be copied as well.
+
+Several directories then need to be mounted as follows:
+
+```
+# mount -t proc none <chroot_dir>/proc
+# mount -t sysfs none <chroot_dir>/sys
+# mount --rbind /dev <chroot_dir>/dev
+# mount --rbind /run <chroot_dir>/run
+```
+
+Use [chroot(1)](https://man.voidlinux.org/chroot.1) to change to the new root,
+then run programs and do tasks as usual. Once you've finished using it, unmount
+the chroot using [umount(8)](https://man.voidlinux.org/umount.8).
```

Comment:
Can we add a warning about removing chroot when not everything is unmounted and what to do in that case?